packages/compartment-mapper/src/bundle.js

```diff
@@ -56,7 +67,10 @@ const sortedModules = (
     if (source) {
       const { record, parser } = source;
       if (record) {
-        const { imports = [], reexports = [] } = record;
+        const {
+          imports = [],
+          reexports = [],
+        } = /** @type {PrecompiledStaticModuleInterface} */ (record);
```

Comment:
Probably dumb question, but there's no way the `record` is a `ThirdPartyStaticModuleInterface` here?
I wasn't sure originally why the casting was necessary.

Reply:
Yeah, the necessity of casting perplexed me too, since `record` in this position can be either a `PrecompiledStaticModuleInterface` or `ThirdPartyStaticModuleInterface`, but only the former has a `reexports`, and the default `[]` is sufficient in all cases when `record` is 3rd party. This is trickery, but it works for both cases because of the default.

Reply:
Yeah the problem is you cannot access a property that doesn't exist in one of the types in a union. An alternative trick might be to specify the `ThirdPartyStaticModuleInterface` differently: 

```
export interface ThirdPartyStaticModuleInterface {
  imports: Array<string>;
  exports: Array<string>;
  reexports?: void;
  execute(proxiedExports: Object, compartment: Compartment, resolvedImports: Record<string, string>): void;
}
```